hack/jenkins/e2e-runner.sh

```diff
@@ -328,7 +328,9 @@ fi
 # * started and destroyed (normal e2e)
 # * neither started nor destroyed (soak test)
 if [[ "${E2E_UP:-}" == "${E2E_DOWN:-}" && -f "${gcp_resources_before}" && -f "${gcp_resources_after}" ]]; then
-  if ! diff -sw -U0 -F'^\[.*\]$' "${gcp_resources_before}" "${gcp_resources_after}" && [[ "${FAIL_ON_GCP_RESOURCE_LEAK:-}" == "true" ]]; then
+  difference=$(diff -sw -U0 -F'^\[.*\]$' "${gcp_resources_before}" "${gcp_resources_after}") || true
+  if [[ -n $(echo "${difference}" | tail -n +3 | grep -E "^\+") ]] && [[ "${FAIL_ON_GCP_RESOURCE_LEAK:-}" == "true" ]]; then
+    echo $difference
```

Comment:
yeah, quotes look very necessary for making this readable:

```shell
$ difference=$(diff -sw -U0 -F'^\[.*\]$' before.txt after.txt)
$ echo ${difference}
--- before.txt 2016-03-29 15:57:44.791933894 -0700 +++ after.txt 2016-03-29 15:58:17.067963690 -0700 @@ -31,0 +32 @@ [ routes ] +default-route-2bbf11db8c66173c jenkins-e2e 0.0.0.0/0 default-internet-gateway 1000 @@ -33,0 +35 @@ [ routes ] +default-route-bff17f6567323330 jenkins-e2e 10.240.0.0/16 1000 @@ -43,0 +46,2 @@ [ firewall-rules ] +jenkins-e2e-default-internal jenkins-e2e 10.0.0.0/8 tcp:1-65535,udp:1-65535,icmp +jenkins-e2e-default-ssh jenkins-e2e 0.0.0.0/0 tcp:22
$ echo "${difference}"
--- before.txt  2016-03-29 15:57:44.791933894 -0700
+++ after.txt   2016-03-29 15:58:17.067963690 -0700
@@ -31,0 +32 @@ [ routes ]
+default-route-2bbf11db8c66173c jenkins-e2e 0.0.0.0/0     default-internet-gateway 1000
@@ -33,0 +35 @@ [ routes ]
+default-route-bff17f6567323330 jenkins-e2e 10.240.0.0/16                          1000
@@ -43,0 +46,2 @@ [ firewall-rules ]
+jenkins-e2e-default-internal jenkins-e2e 10.0.0.0/8   tcp:1-65535,udp:1-65535,icmp
+jenkins-e2e-default-ssh      jenkins-e2e 0.0.0.0/0    tcp:22
```

I love shell so much